runtime/jvmti/jvmtiHook.c

```diff
@@ -3041,16 +3053,16 @@ jvmtiHookCompilingEnd(J9HookInterface** hook, UDATA eventNum, void* eventData, v
 static void 
 jvmtiHookSampledObjectAlloc(J9HookInterface** hook, UDATA eventNum, void* eventData, void* userData)
 {
-	J9JVMTIEnv * j9env = userData;
+	J9JVMTIEnv *j9env = userData;
+	MM_ObjectAllocationSamplingEvent *data = eventData;
 	jvmtiEventSampledObjectAlloc callback = j9env->callbacks.SampledObjectAlloc;
+	J9VMThread *currentThread = data->currentThread;
```

Comment:
Since `data` is only used once.
```suggestion
	J9VMThread *currentThread = ((MM_ObjectAllocationSamplingEvent *)data)->currentThread;
```